src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/matchtoterm/TermFieldRewriter.java

```diff
@@ -210,19 +214,52 @@ public boolean isValidIdentifierForTerm(SQLIdentifierExpr expr) {
             );
     }
 
-    public boolean hasIdenticalMappings(TermFieldScope scope, Map<String, String> indexToType) {
+    private void checkMappingCompatibility(TermFieldScope scope, Map<String, String> indexToType) {
         if (scope.getMapper().isEmpty()) {
             throw new VerificationException("Unknown index " + indexToType.keySet());
         }
 
-        if (isMappingOfAllIndicesDifferent()) {
-            return false;
-        }
+        Set<FieldMappings> mappingSet = curScope().getMapper().allMappings().stream().
+                                 flatMap(typeMappings -> typeMappings.allMappings().stream()).
+                                 collect(Collectors.toSet());
+
+        FieldMappings fieldMappings = null;
 
+        if (mappingSet.size() > 1) {
+            Map<String, Map<String, Object>> mergedMapping = new HashMap<>();
+
+            for (FieldMappings f : mappingSet) {
+                Map<String, Map<String, Object>> m = f.data();
+                m.forEach((k, v) -> getMergedMapping(k, v, mergedMapping));
+            }
+
+            fieldMappings = new FieldMappings(mergedMapping);
+        } else {
+            fieldMappings = curScope().getMapper().firstMapping().firstMapping();
+        }
         // We need finalMapping to lookup for rewriting
-        FieldMappings fieldMappings = curScope().getMapper().firstMapping().firstMapping();
         curScope().setFinalMapping(fieldMappings);
-        return true;
+    }
+
+    private void getMergedMapping(String fieldName, Map<String, Object> fieldMappingValue , Map<String, Map<String, Object>> mergedMapping){
```

Comment:
Sorry, missed this one, if the function name is `getSomething`, then it should return something, maybe `verifySingleFieldMapping` or whatever better name you think of.

Reply:
I was also thinking about this. The method is actually constructing the mapping we want to use. It's adding content to the Map `mergedMapping`, which is passed as a parameter.  Does `mergeMapping` sound like a good method name to you?

Reply:
I think adding a new mapping to that merged map is just a side effect of this method, not the main function, so I am more inclined to assert the verification part in the name.

Reply:
Got you. Fixed as `verifySingleFieldMapping`